### ISSUE TITLE
Kubernetes related issue while installing OIH. Need some quick help troubleshooting.

### ISSUE THREAD

**Issue by ricardopieper** [openintegrationhub/openintegrationhub#930]
Hi,
I'm trying to install OIH on my laptop using minikube by following the tutorial. I am also following the tutorials on minikube and kubectl, and so far it seems to be working fine, even the minikube dashboard works.

This is my first time using kubernetes as well. I have some limited experience with docker swarm and its networking concepts.

I executed `kubectl apply -f ./1-Platform` and everything worked fine. Looks like this step just starts up rabbitmq, mongodb and redis.

Then I tried `kubectl apply -f ./2-IAM`. However, the pod fails. This is the log:

```
{"logger":"check-env","level":"warn","message":"Missing IAM_ACC_SERVICEACCOUNT_PASSWORD. This password will be auto-generated only displayed once via stdout!\n\n    ===============================================\n\n    || Auto generated password: fnVnUOy9y3WUEkrg ||\n\n    ===============================================\n"}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"err":{"message":"connect ECONNREFUSED 127.0.0.1:5672","name":"Error","stack":"Error: connect ECONNREFUSED 127.0.0.1:5672\n    at Object._errnoException (util.js:1022:11)\n    at _exceptionWithHostPort (util.js:1044:20)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1198:14)","code":"ECONNREFUSED"},"msg":"connect ECONNREFUSED 127.0.0.1:5672","time":"2020-02-01T19:20:07.100Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"msg":"An error occurred. Starting reconnecting","time":"2020-02-01T19:20:07.100Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"err":{"message":"connect ECONNREFUSED 127.0.0.1:5672","name":"Error","stack":"Error: connect ECONNREFUSED 127.0.0.1:5672\n    at Object._errnoException (util.js:1022:11)\n    at _exceptionWithHostPort (util.js:1044:20)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1198:14)","code":"ECONNREFUSED"},"msg":"connect ECONNREFUSED 127.0.0.1:5672","time":"2020-02-01T19:20:08.104Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"msg":"An error occurred. Starting reconnecting","time":"2020-02-01T19:20:08.104Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"err":{"message":"connect ECONNREFUSED 127.0.0.1:5672","name":"Error","stack":"Error: connect ECONNREFUSED 127.0.0.1:5672\n    at Object._errnoException (util.js:1022:11)\n    at _exceptionWithHostPort (util.js:1044:20)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1198:14)","code":"ECONNREFUSED"},"msg":"connect ECONNREFUSED 127.0.0.1:5672","time":"2020-02-01T19:20:09.107Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"msg":"An error occurred. Starting reconnecting","time":"2020-02-01T19:20:09.107Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"err":{"message":"connect ECONNREFUSED 127.0.0.1:5672","name":"Error","stack":"Error: connect ECONNREFUSED 127.0.0.1:5672\n    at Object._errnoException (util.js:1022:11)\n    at _exceptionWithHostPort (util.js:1044:20)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1198:14)","code":"ECONNREFUSED"},"msg":"connect ECONNREFUSED 127.0.0.1:5672","time":"2020-02-01T19:20:10.109Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"msg":"An error occurred. Starting reconnecting","time":"2020-02-01T19:20:10.109Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"err":{"message":"connect ECONNREFUSED 127.0.0.1:5672","name":"Error","stack":"Error: connect ECONNREFUSED 127.0.0.1:5672\n    at Object._errnoException (util.js:1022:11)\n    at _exceptionWithHostPort (util.js:1044:20)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1198:14)","code":"ECONNREFUSED"},"msg":"connect ECONNREFUSED 127.0.0.1:5672","time":"2020-02-01T19:20:11.112Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"msg":"An error occurred. Starting reconnecting","time":"2020-02-01T19:20:11.112Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"err":{"message":"connect ECONNREFUSED 127.0.0.1:5672","name":"Error","stack":"Error: connect ECONNREFUSED 127.0.0.1:5672\n    at Object._errnoException (util.js:1022:11)\n    at _exceptionWithHostPort (util.js:1044:20)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1198:14)","code":"ECONNREFUSED"},"msg":"connect ECONNREFUSED 127.0.0.1:5672","time":"2020-02-01T19:20:12.115Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"msg":"An error occurred. Starting reconnecting","time":"2020-02-01T19:20:12.115Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"err":{"message":"connect ECONNREFUSED 127.0.0.1:5672","name":"Error","stack":"Error: connect ECONNREFUSED 127.0.0.1:5672\n    at Object._errnoException (util.js:1022:11)\n    at _exceptionWithHostPort (util.js:1044:20)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1198:14)","code":"ECONNREFUSED"},"msg":"connect ECONNREFUSED 127.0.0.1:5672","time":"2020-02-01T19:20:13.116Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"msg":"An error occurred. Starting reconnecting","time":"2020-02-01T19:20:13.116Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"err":{"message":"connect ECONNREFUSED 127.0.0.1:5672","name":"Error","stack":"Error: connect ECONNREFUSED 127.0.0.1:5672\n    at Object._errnoException (util.js:1022:11)\n    at _exceptionWithHostPort (util.js:1044:20)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1198:14)","code":"ECONNREFUSED"},"msg":"connect ECONNREFUSED 127.0.0.1:5672","time":"2020-02-01T19:20:14.118Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"msg":"An error occurred. Starting reconnecting","time":"2020-02-01T19:20:14.119Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"err":{"message":"connect ECONNREFUSED 127.0.0.1:5672","name":"Error","stack":"Error: connect ECONNREFUSED 127.0.0.1:5672\n    at Object._errnoException (util.js:1022:11)\n    at _exceptionWithHostPort (util.js:1044:20)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1198:14)","code":"ECONNREFUSED"},"msg":"connect ECONNREFUSED 127.0.0.1:5672","time":"2020-02-01T19:20:15.141Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"msg":"An error occurred. Starting reconnecting","time":"2020-02-01T19:20:15.141Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"err":{"message":"connect ECONNREFUSED 127.0.0.1:5672","name":"Error","stack":"Error: connect ECONNREFUSED 127.0.0.1:5672\n    at Object._errnoException (util.js:1022:11)\n    at _exceptionWithHostPort (util.js:1044:20)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1198:14)","code":"ECONNREFUSED"},"msg":"connect ECONNREFUSED 127.0.0.1:5672","time":"2020-02-01T19:20:16.184Z","v":0}
{"name":"accounts","hostname":"iam-646dd69578-2hvt5","pid":17,"level":50,"msg":"An error occurred. Starting reconnecting","time":"2020-02-01T19:20:16.184Z","v":0}
{"message":"failed to connect to server [mongodb-service.oih-dev-ns.svc.cluster.local:27017] on first connect [MongoNetworkError: getaddrinfo EAI_AGAIN mongodb-service.oih-dev-ns.svc.cluster.local:27017]","stack":"MongoNetworkError: failed to connect to server [mongodb-service.oih-dev-ns.svc.cluster.local:27017] on first connect [MongoNetworkError: getaddrinfo EAI_AGAIN mongodb-service.oih-dev-ns.svc.cluster.local:27017]\n    at Pool.<anonymous> (/tmp/node_modules/mongoose/node_modules/mongodb-core/lib/topologies/server.js:431:11)\n    at emitOne (events.js:116:13)\n    at Pool.emit (events.js:211:7)\n    at connect (/tmp/node_modules/mongoose/node_modules/mongodb-core/lib/connection/pool.js:557:14)\n    at makeConnection (/tmp/node_modules/mongoose/node_modules/mongodb-core/lib/connection/connect.js:39:11)\n    at callback (/tmp/node_modules/mongoose/node_modules/mongodb-core/lib/connection/connect.js:261:5)\n    at Socket.err (/tmp/node_modules/mongoose/node_modules/mongodb-core/lib/connection/connect.js:286:7)\n    at Object.onceWrapper (events.js:315:30)\n    at emitOne (events.js:116:13)\n    at Socket.emit (events.js:211:7)\n    at emitErrorNT (internal/streams/destroy.js:64:8)\n    at _combinedTickCallback (internal/process/next_tick.js:138:11)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","name":"MongoNetworkError","errorLabels":["TransientTransactionError"],"logger":"accounts/init","level":"error"}
{"logger":"accounts/init","level":"info","message":"Clean shutdown"}
{"logger":"accounts/init","level":"error","message":"error"}
```

So the issue seems to be that the application in the IAM pod cannot reach mongodb and rabbitmq in the other pods. In the mongodb case, it is even trying to resolve the internal DNS name.Seems to be the correct address:

```
/usr/src/app $ echo $IAM_MONGODB_CONNECTION
mongodb://mongodb-service.oih-dev-ns.svc.cluster.local/iam
```

My laptop is:
 - Elementary OS 5.1 (debian based). Fresh install, SSD formatted from scratch, just to try OIH.
 - kubernetes (1.17.2), docker from apt-get install docker (as recommended by kubectl installation guide IIRC), latest minikube.
 - Recently cloned OIH (commit 58b4a4c636d231315a4e4e31b8a61f3e95502455)
 - cat /etc/hosts:
```
::1     ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
192.168.0.6 iam.localoih.com smk.localoih.com flow-repository.localoih.com auditlog.localoih.com metadata.localoih.com component-repository.localoih.com webhooks.localoih.com attachment-storage-service.localoih.com data-hub.localoih.com ils.localoih.com web-ui.localoih.com
```

I double-checked mongodb's pod logs to see if it is online. Apparently it is.

Can you help me figuring out why I'm having this issue? The installation guide does not say anything about it.

**Comment by RobinBrinkmann** [comment]
@SvenHoeffler and @sachmerz please have a look!

**Comment by SvenHoeffler** [comment]
Hi @ricardopieper , you are correct that the IAM apparently can't reach the Mongo DB. I'm currently trying to reproduce the error, but have not yet managed it. In the meantime, you could try the following: 

- Force recreate the IAM pod. To do that, in the minikube dashboard, navigate to "Pods" and find the IAM's current pod (It will be named `iam-` followed by a randomised ID). If you can't find it, make sure the namespace "oih-dev-ns" is selected. Then simply delete the pod, and the workload should recreate a new one. Alternatively, you can go to the workload directly and scale it down to 0 replicas, wait until the pod is destroyed, and then scale it back up to 1 replica.
- In minikube dashboardm navigate to "Persistent Volumes" and check whether any errors are shown for in the local volume created for the Mongo DB.

**Comment by ricardopieper** [comment]
Hi... unfortunately my laptop SSD broke like 5 minutes ago, so I cannot continue testing this right now. 

However, I had checked what you asked but forgot to mention in the original post. I deleted the pod several times and it recreated by itself. I also tried reinstalling the whole thing. 

The database seemed okay, no erros in the mongodb logs, however I did not check whether the volumes were ok... maybe it wasn't, since my SSD ended up failing. 

The connection to RabbitMQ was failing for the same reason (connection refused). So I don't think the issue was exclusive to mongodb.

In a couple days I'll try fixing the laptop and recover the installation/retry from scratch, but maybe it was doomed to fail from the beginning due to a hardware failure. Though there isn't much evidence of that.
